### PR TITLE
LPS-145630 Allow do not cache files based in the mime type

### PR DESCRIPTION
**Pull request by robertoDiaz**

### CODE REVIEW

**Comment by robertoDiaz** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/servlet/CacheControlFileEntryHttpHeaderCustomizer.java [hunk 1]
```diff
@@ -79,6 +80,13 @@ private String _getHttpHeaderValue(FileEntry fileEntry, String currentValue)
 			return currentValue;
 		}
 
+		if (ArrayUtil.contains(
+				_cacheControlConfiguration.noCacheableMimeTypes(),
+				fileEntry.getMimeType())) {
+
+			return HttpHeaders.CACHE_CONTROL_NO_CACHE_VALUE;
+		}
+
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/servlet/CacheControlFileEntryHttpHeaderCustomizer.java:89]: Shouldn't we move this block of code to precede the permission check? The "no-cache" policy is even more strict than the default policy (private); to me, it makes sense to let admins make things more restrictive.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/servlet/CacheControlFileEntryHttpHeaderCustomizer.java (hunk 1 above)
Comment (robertoDiaz) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/servlet/CacheControlFileEntryHttpHeaderCustomizer.java:89]: yes, you are right, we do not need to check permission before the no-cache policy.

I'll resend :)

**Comment by robertoDiaz** [comment]
Hey @adolfopa , I've updated the PR

**Comment by robertoDiaz** [comment]
ci:test:sf

**Comment by robertoDiaz** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by robertoDiaz** [comment]
@adolfopa it seems that tests are stuck,

Can you push it to master (we need to backport this to fix a performance issue)

Thanks!

**Comment by adolfopa** [comment]
ci:stop

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

**Comment by robertoDiaz** [comment]
Closing to rebase and resend

**Pull request closed without merging**